sdkit/utils/memory_utils.py

```diff
@@ -9,94 +12,141 @@
+            after calling `gc()` (for garbage-collection).
 
-    For e.g. you can record variables from across the codebase, and see which one is leaking by calling
-    `print_largest_tensors_in_memory()` or `take_memory_snapshot()` after calling `gc()` (for garbage-collection).
-    
-    `print_largest_tensors_in_memory()` and `take_memory_snapshot()` print the recorded names for each tensor, if available.
+    `print_largest_tensors_in_memory()` and `take_memory_snapshot()`
+        print the recorded names for each tensor, if available.
     '''
 
     obj_id = get_object_id(t)
-    recorded_tensor_names[obj_id] = [] if obj_id not in recorded_tensor_names else recorded_tensor_names[obj_id]
+    recorded_tensor_names[obj_id] = [] if obj_id not in recorded_tensor_names \
+        else recorded_tensor_names[obj_id]
     recorded_tensor_names[obj_id].append(name)
 
-    if log_info: print_tensor_info(t, name)
+    if log_info:
+        print_tensor_info(t, name)
+
 
 def print_tensor_info(t, name='t'):
-    'Prints a summary of the tensor, for e.g. its size, shape, data type, device etc.'
+    '''
+        Prints a summary of the tensor,
+          for e.g. its size, shape, data type, device etc.
+    '''
 
     from sdkit.utils import log
 
     obj_id = get_object_id(t)
-    obj_size = t.nelement() * t.element_size() / 1024**2 # MiB
-    log.info(f' {name} id: {obj_id}, size: {obj_size} MiB, shape: {t.shape}, requires_grad: {t.requires_grad}, type: {t.dtype}, device: {t.device}')
+    obj_size = t.nelement() * t.element_size() / 1024**2  # MiB
+    log.info(
+        ' %s id: %s, size: %s MiB, shape: %s, '
+        'requires_grad: %s, type: %s, device: %s',
+        name, obj_id, obj_size, t.shape, t.requires_grad, t.dtype, t.device
+    )
+
 
 def get_tensors_in_memory(device):
     '''
     Returns the list of all the tensor objects in memory, on the given device.
-    **Warning: Do not keep a reference to the returned list longer than necessary, since that will
-    prevent garbage-collection of all the tensors in memory.**
+    **  Warning: Do not keep a reference to the returned list longer than
+    **  necessary, since that will prevent garbage-collection of all the
+    **  tensors in memory.
     '''
+    from sdkit.utils import log
 
     device = torch.device(device) if isinstance(device, str) else device
     tensors = []
     objs_in_mem = get_objects()
     for obj in objs_in_mem:
         try:
-            if (torch.is_tensor(obj) or (hasattr(obj, 'data') and torch.is_tensor(obj.data))) and obj.device == device:
+            if (
+                torch.is_tensor(obj) or
+                (
+                    hasattr(obj, 'data') and
+                    torch.is_tensor(obj.data)
+                )
+            ) and obj.device == device:
                 tensors.append(obj)
-        except:
-            pass
+        except Exception as ex:
+            log.warning('Error while getting tensors in memory: %s', ex)
```

Comment:
This `pass` statement is intentional. Lots of objects in the memory will throw errors, so this function will be unusable otherwise. https://discuss.pytorch.org/t/how-to-debug-causes-of-gpu-memory-leaks/6741/3